cpp/include/cudf/strings/detail/utilities.cuh

```diff
@@ -195,8 +195,17 @@ class per_context_cache {
   template <typename Initializer>
   TableType* find_or_initialize(const Initializer& init)
   {
+    void* cuda_lib                                 = nullptr;
+    decltype(cuCtxGetCurrent)* context_get_current = nullptr;
+
+    cuda_lib = dlopen("libcuda.so", RTLD_LAZY | RTLD_LOCAL | RTLD_NODELETE);
+    CUDF_EXPECTS(cuda_lib != nullptr, "Failed to load cuda library");
+    context_get_current =
+      reinterpret_cast<decltype(context_get_current)>(dlsym(cuda_lib, "cuCtxGetCurrent"));
+    CUDF_EXPECTS(context_get_current != nullptr, "could not find cuCtxGetCurrent symbol");
```

Comment:
I strongly suspect we can eliminate the need for using the driver API at all here by just using `cudaGetDevice` instead.

The only reason for using `cuCtxGetCurrent` would be if someone is manually constructing and managing a custom `CUcontext` and I strongly suspect no one using libcudf is doing that. The exception could be Legion/Realm given @magnatelee is the one who originally wrote this here: https://github.com/rapidsai/cudf/issues/5335

Reply:
@jrhemstad I think that code should work equally well with `cudaGetDevice`.

Reply:
Thanks Wonchan Lee. I can work on changing this code to use `cudaGetDevice` instead in a separate PR to not hold up this fix. Would that be ok @jrhemstad ?

Reply:
If you just want to take over  this PR @davidwendt , that  would be great as well!